include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -100,6 +100,21 @@ class MGConstrainedDoFs : public Subscriptor
     const std::set<types::boundary_id> &boundary_ids,
     const ComponentMask &               component_mask = ComponentMask());
 
+  /**
+   * Fill the internal data structures with information
+   * about no normal flux boundary dofs.
+   *
+   * This function is limited to meshes whose no normal flux boundaries
+   * have faces which are normal to the x, y, or z axis. Also, when generating
+   * the mesh, the user must have set colorize=true and call
```

Comment:
Here, and in the assertion you have in the code, you seem to have a very specific way in mind how someone would create such a mesh. But that's not necessarily true -- one could have read such a mesh from a file, or used a different function in `namespace GridGenerator`, and neither may have a `colorize` option.

I think a better way is to state the documentation in terms of the condition you want to be true at the time this function is called, not something that would have been true at some point in the past. And then say something like "One way how this condition can be true is if, for example, the mesh used here was produced using GridGenerator::... using the `colorize=true` option."

Reply:
Yes, good point. Let me know if the new wording is more clear.